lib/datadog/tracing/span_event.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'time'
+
+module Datadog
+  module Tracing
+    # SpanEvent represents an annotation on a span.
+    # @public_api
+    class SpanEvent
+      # @!attribute [r] name
+      #   @return [Integer]
+      attr_reader :name
+
+      # @!attribute [r] attributes
+      #   @return [Hash{String => String, Numeric, Boolean, Array<String, Numeric, Boolean>}]
+      attr_reader :attributes
+
+      # @!attribute [r] time_unix_nano
+      #   @return [Integer]
+      attr_reader :time_unix_nano
+
+      def initialize(
+        name,
+        attributes: nil,
+        time_unix_nano: nil
```

Comment:
I see that the time is eventually exported in nanoseconds, but in Ruby most of the times are kept in (floating-point) seconds. If it is desired to keep the time in nanoseconds it would be helpful to have a note explaining the reasoning for this decision.

Reply:
Initially this was done for simplicy. Timestamps for span events are serialized in nanoseconds. There's no need to store them in seconds. 

I believe the start time and duration of a span is stored in seconds. These timestamps are converted to nanoseconds when a span is serialized and sent to the agent. I can use the same approach here.